platform/ios/app/MBXState.m

```diff
@@ -0,0 +1,77 @@
+#import "MBXState.h"
+
+NSString *const MBXCamera = @"MBXCamera";
+NSString *const MBXUserTrackingMode = @"MBXUserTrackingMode";
+NSString *const MBXShowsUserLocation = @"MBXShowsUserLocation";
+NSString *const MBXDebugMask = @"MBXDebugMask";
+NSString *const MBXShowsZoomLevelHUD =  @"MBXShowsZoomLevelHUD";
+NSString *const MBXShowsTimeFrameGraph = @"MBXShowsFrameTimeGraph";
+
+
+@implementation MBXState
+
+- (instancetype) init {
+    if ((self = super.self)) {
+        NSMutableDictionary *dictionary = [NSMutableDictionary dictionaryWithObjectsAndKeys: @"", MBXCamera,
+            @"", MBXUserTrackingMode,
+            @"", MBXShowsUserLocation,
+            @"", MBXDebugMask,
+            @"", MBXShowsZoomLevelHUD,
+            @"", MBXShowsTimeFrameGraph,
+            nil];
+
+        _state = dictionary;
+    }
+    return self;
+}
+
+- (void)saveMapCameraState:(MGLMapCamera *)camera {
```

Comment:
I think it would be better if `MBXState` was a simple class that could be `initWithCoder`'d, and leave the state saving/restoration to `MBXStateManager`. `MBXState` doesn't need to know (or care) how it's being saved.

If you move the load/save logic, then I think you can lose the dictionary (unless you have plans for that?).